jena-arq/src/main/java/org/apache/jena/sparql/function/library/FN_Apply.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jena.sparql.function.library;
+
+import java.util.List ;
+
+import org.apache.jena.atlas.lib.Cache ;
+import org.apache.jena.atlas.lib.CacheFactory ;
+import org.apache.jena.graph.Node ;
+import org.apache.jena.sparql.expr.ExprEvalException ;
+import org.apache.jena.sparql.expr.ExprList ;
+import org.apache.jena.sparql.expr.NodeValue ;
+import org.apache.jena.sparql.function.Function ;
+import org.apache.jena.sparql.function.FunctionBase ;
+import org.apache.jena.sparql.function.FunctionFactory ;
+import org.apache.jena.sparql.function.FunctionRegistry ;
+import org.apache.jena.sparql.sse.builders.ExprBuildException ;
+import org.apache.jena.sparql.util.Context ;
+
+/** XPath and XQuery Functions and Operators 3.1
+ * <p> 
+ * {@code fn:apply(function, args)}
+ */
+public class FN_Apply extends FunctionBase {
+    // Assumes one object per use site. 
+    private Cache<String, Function> cache1 = CacheFactory.createOneSlotCache();
+    
+    @Override
+    public void checkBuild(String uri, ExprList args) {
+        if ( args.isEmpty() )
+            throw new ExprBuildException("fn:apply: no function to call (minimum number of args is one)");
+    }
+    @Override
+    public NodeValue exec(List<NodeValue> args) {
+        if ( args.isEmpty() )
```

Comment:
Ideally, but no harm checking. `build` is only guaranteed to happen when called from parsed code. The build pass is a query thing to move errors from runtime to setup time. Otherwise, you get many, many runtime errors.